Sources/App/Onboarding/Screens/OnboardingServersList/ManualURLEntry/ManualURLEntryView.swift

```diff
@@ -0,0 +1,102 @@
+import Shared
+import SwiftUI
+
+struct ManualURLEntryView: View {
+    @Environment(\.dismiss) private var dismiss
+    @State private var urlString = ""
+    @FocusState private var focused: Bool?
+    @State private var showInvalidURLError = false
+
+    let connectAction: (URL) -> Void
+
+    var body: some View {
+        NavigationView {
+            List {
+                Section(L10n.Onboarding.ManualSetup.TextField.title) {
+                    TextField(L10n.Onboarding.ManualSetup.TextField.placeholder, text: $urlString)
+                        .keyboardType(.URL)
+                        .focused($focused, equals: true)
+                        .onAppear {
+                            focused = true
+                        }
+                }
+
+                httpOrHttpsSection
+            }
+            .navigationTitle(L10n.Onboarding.ManualSetup.title)
+            .navigationViewStyle(.stack)
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .topBarTrailing) {
+                    CloseButton {
+                        dismiss()
+                    }
+                }
+            }
+            .safeAreaInset(edge: .bottom) {
+                connectButton
+            }
+            .alert(isPresented: $showInvalidURLError) {
+                Alert(
+                    title: Text(verbatim: L10n.Onboarding.ManualSetup.InputError.title),
+                    message: Text(verbatim: L10n.Onboarding.ManualSetup.InputError.message),
+                    dismissButton: .default(Text(verbatim: L10n.okLabel))
+                )
+            }
+        }
+    }
+
+    @ViewBuilder
```

Comment:
I would add a comment on this function I had to run the app to understand what was it all about.